librz/arch/isa/mips/il/mips32.c

```diff
@@ -0,0 +1,1435 @@
+#define MIPS32_LINK()         SETG(MIPS32_REG_RA, UN(MIPS32_GPRLEN, insn->address + 8))
+
+static RzILOpEffect *mips32_il_move(csh *handle, cs_insn *insn) {
+	const char *rt = REG(0);
+	RzILOpPure *val = NULL;
+
+	if (REG_IS_ZERO(1)) {
+		// set zero if target register is $zero
+		val = SN(MIPS32_GPRLEN, 0);
+	} else if (IS_IMM(1)) {
+		val = MIPS32_IMM(2);
+	} else {
+		val = MIPS32_REG(1);
+	}
+
+	return SETG(rt, val);
+}
+
+static RzILOpEffect *mips32_il_abs_fmt(csh *handle, cs_insn *insn) {
+	// Floating Point Absolute Value
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_add(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *rt = MIPS32_REG(2);
+
+	RzILOpPure *sum = SIGNED(MIPS32_GPRLEN, ADD(rs, rt));
+	RzILOpPure *overflow = CHECK_OVERFLOW(DUP(sum), 32);
+	return BRANCH(overflow, IL_CAUSE_OVERFLOW(), SETG(rd, sum));
+}
+
+static RzILOpEffect *mips32_il_add_fmt(csh *handle, cs_insn *insn) {
+	// Floating Point Add
+	return NULL;
+}
+
+static RzILOpEffect *mips32_il_addi(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *imm = MIPS32_IMM(2);
+
+	RzILOpPure *sum = SIGNED(MIPS32_GPRLEN, ADD(rs, imm));
+	RzILOpPure *overflow = CHECK_OVERFLOW(DUP(sum), 32);
+	return BRANCH(overflow, IL_CAUSE_OVERFLOW(), SETG(rd, sum));
+}
+
+static RzILOpEffect *mips32_il_addiu(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *imm = MIPS32_IMM(2);
+	RzILOpPure *sum = ADD(rs, imm);
+	return SETG(rd, sum);
+}
+
+static RzILOpEffect *mips32_il_addu(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *rt = MIPS32_REG(2);
+	RzILOpPure *sum = ADD(rs, rt);
+	return SETG(rd, sum);
+}
+
+static RzILOpEffect *mips32_il_and(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *rt = MIPS32_REG(2);
+	RzILOpPure *sum = LOGAND(rs, rt);
+	return SETG(rd, sum);
+}
+
+static RzILOpEffect *mips32_il_andi(csh *handle, cs_insn *insn) {
+	if (REG_IS_ZERO(0)) {
+		return NOP(); // return NOP if target register is $zero
+	}
+
+	const char *rd = REG(0);
+	RzILOpPure *rs = MIPS32_REG(1);
+	RzILOpPure *imm = MIPS32_IMM(2);
+	RzILOpPure *sum = LOGAND(rs, imm);
+	return SETG(rd, sum);
+}
+
+static RzILOpEffect *mips32_il_b(csh *handle, cs_insn *insn) {
+	RzILOpPure *target = MIPS32_IMM(0);
+	return JMP(target);
+}
+
+static RzILOpEffect *mips32_il_bal(csh *handle, cs_insn *insn) {
```

Comment:
```suggestion
static RzILOpEffect *mips32_il_bal(const csh *handle, const cs_insn *insn) {
```

Is this possible? Then it would be nice to have it in all of them.

Reply:
Yes I can convert this into const. I also need to add an argument for the size of the regs.